### PR TITLE
feat(STONEINTG-901): added debugging page for integration tests

### PR DESCRIPTION
**Pull request by chipspeak**
- Added a new page for debugging integration tests via console
- Added supporting screenshots
- Updated nav accordingly
- Tested via local build

### CODE REVIEW

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc [hunk 1]
```diff
@@ -0,0 +1,44 @@
+= Debugging Integration Tests
+
+In the event of an integration test scenario failing, there are a number of steps to take in order to debug the issue before rerunning the test in question. These steps are outlined below.
+
+
+.Prerequisites
+- You have a snapshot that has completed all its initial tests. All initial tests must be finished before attempting to debug the failed test.
+
+- You have access to the console in order to view the logs of the failed test.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:9]: I think that it has been called the `{ProductName} UI` instead of the `console`.

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+.Procedure
+
+. Identify the IntegrationTestScenario that has failed. This can be found in the console by navigating to 
+Applications/your_application_name/activity and select "Pipeline Runs" under "Activity By". Your view should be similar to the below image.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:14]: Would it make sense to separate these steps out similar to how the verification steps have been listed for other integration test pages?

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
++
+image::activityview.png[role="border" alt="Activity View"]
+
+. Click on the pipeline run that has a status of "Failed". This will take you to a detailed view of the failed run. On the right side of this page, look for the "Log Snippet" section. This will contain the most recent error returned by the integration test. For more comprehensive logs, click on "See Logs". See the below image for reference.
+
++
+image::testfailed.png[role="border" alt="Test Failed"]
+
+. In the above example the output suggests that the pipelineRun is missing an expected parameter, "username". To edit the integration test, scroll down to the "Integration Test" field and click on the listed test. This should take you to a view similar to the one seen below.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:24]: Are there different types of failure mechanisms? Are you only trying to show users how to debug and resolve missing parameter issues?

I feel like this needs to be more generic.

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
++
+image::testview.png[role="border" alt="Test View"]
+
+. Click on the "Actions" button seen above and select "Edit". This should take you to the below view.
+
++
+image::edittest.png[role="border" alt="Edit Test"]
+
+. In this example we will need to click on the "Parameters" dropdown and add the parameter expected by the pipeline run.
+In this case it's "username" and we will set the value to "testuser". Once you've completed your edits, click "Save Changes" at the bottom of the screen to update the integration test.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:35]: This is at least partially invalidated by the recent PR #92. Can we instead integrate that in here and/or add additional information to that page related to these changes?

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+
+. Click on the "Actions" button seen above and select "Edit". This should take you to the below view.
+
++
+image::edittest.png[role="border" alt="Edit Test"]
+
+. In this example we will need to click on the "Parameters" dropdown and add the parameter expected by the pipeline run.
+In this case it's "username" and we will set the value to "testuser". Once you've completed your edits, click "Save Changes" at the bottom of the screen to update the integration test. 
+
+. Return to the activity view mentioned in step 1 and click on the 3 verticle dots on the right side of the pipeline run. Within this expanded menu, click "rerun". This will rerun the integration test with our updated parameter. This should result in a success as seen in the below image. For a more comprehensive, CLI-focused explanation of rerunning tests, click xref:/how-tos/testing/integration/rerunning.adoc[here].
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:37]: Can we instead add information on how to rerun from the UI to https://konflux-ci.dev/docs/how-tos/testing/integration/rerunning/? Then we can just simply link users there to rerun the test.

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+
++
+image::successfulretest.png[role="border alt="Successful Retest"]
+
++
+NOTE: This outlines a simple approach to debugging and rerunning via the console. For a more comprehensive guide on how to rerun a test via the CLI, please see the next page.
```
Comment (arewm) [inline docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:43]: If you want to make a reference to the next page, you can link to it. I am not sure how much value this note provides though.

**Review by arewm — CHANGES_REQUESTED** [review]

**Comment by Chr1st1anSears** [comment]
@dirgim I added a fellow writer, Oss (@missmesss) as a reviewer. She'll provide a docs review :+1:

**Comment by missmesss** [comment]
@Chr1st1anSears @gtrivedi88 I'll review this one, no worries.

@dirgim Oftentimes writers' reviews come last, after the tech reviews. I see Andrew suggests some changes to the procedures, so I'll keep an eye on this PR and review the changes when they're closer to being finalized, if you don't mind.

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc (hunk 3 above)
Comment (chipspeak) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:24]: The intention was to provide a step-by-step resolution of an example issue but is it the case where broader is better here? I could instead point to the editing page for clarity or else make clearer that this is just one example of an issue and how to resolve it?

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc (hunk 2 above)
Comment (chipspeak) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:14]: Sure thing! Just to make sure I have it right, more like a single sentence per step?

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc (hunk 1 above)
Comment (chipspeak) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:9]: Thanks!

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc (hunk 4 above)
Comment (chipspeak) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:35]: Yeah it feels like the logical thing to do would be to point to link to the editing for the relevant steps and elaborate as needed if that makes sense?

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc (hunk 5 above)
Comment (chipspeak) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:37]: Yeah that makes total sense. I can add the supporting text to those steps as an alternate approach to CLI. I was wondering though, is there an argument in situations like this for a separate subpage for CLI and UI respectively? To me that would provide a bit more clarity for a user.

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc (hunk 6 above)
Comment (chipspeak) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:43]: I'll remove it cheers!

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc [hunk 7]
```diff
@@ -0,0 +1,44 @@
+= Debugging Integration Tests
+
+In the event of an integration test scenario failing, there are a number of steps to take in order to debug the issue before rerunning the test in question. These steps are outlined below.
+
+
+.Prerequisites
+- You have a snapshot that has completed all its initial tests. All initial tests must be finished before attempting to debug the failed test.
```
Comment (ralphbean) [inline docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:7]: What do you mean by "initial tests"? Do you mean the scanners and linters that run as a part of the build pipeline?

I guess "initial tests" is the language used in `docs/modules/ROOT/pages/how-tos/testing/integration/rerunning.adoc`. It's not quite clear to me, but it's probably better to be consistent here than anything else. If we were to revise it, let's do that in a separate PR.

Comment (chipspeak) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:7]: Hi Ralph, I was just adopting the language from the rerunning page and yeah it was my understanding that this was referring to the build pipeline checks!

Comment (arewm) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:7]: We use that term consistently outside of the integration test pages.

Comment (chipspeak) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc]: This is now reflected in the latest commit.

File: .vscode/settings.json [hunk 8]
```diff
@@ -1,4 +1,3 @@
 {
-    "asciidoc.antora.enableAntoraSupport": true,
```
Comment (dirgim) [inline .vscode/settings.json:2]: I think that it's probably best to no change this file as part of this PR

Comment (chipspeak) [reply .vscode/settings.json:2]: ah sorry well spotted!

File: docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc [hunk 9]
```diff
@@ -0,0 +1,36 @@
+= Debugging Integration Tests
+
+In the event of an integration test scenario failing, there are a number of steps to take in order to debug the issue before rerunning the test in question. These steps are outlined below.
+
+
+.Prerequisites
+- You have a snapshot that has completed all built-time tests. These must be completed before attempting to debug the failed test.
```
Comment (dirgim) [inline docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:7]: Looks like it should be `built-time` -> `build-time` based on [the glossary](https://konflux-ci.dev/docs/glossary/).

Comment (chipspeak) [reply docs/modules/ROOT/pages/how-tos/testing/integration/debugging.adoc:7]: Damn it apologies.

**Review by dirgim — APPROVED** [review]
LGTM from my point of view

**Review by arewm — APPROVED** [review]
Thanks!
